configure.ac

```diff
@@ -1015,11 +1015,18 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdint.h>
 
 dnl LevelDB platform checks
 AC_MSG_CHECKING(for fdatasync)
+case $host in
+  *netbsd*)
+AC_MSG_RESULT(no because NetBSD will not fdatasync on directories and this breaks leveldb); HAVE_FDATASYNC=0
```

Comment:
We use `fdatasync` on files in `FileCommit`. Currently, it checks `__linux__ || __NetBSD__`, but it really should be using `HAVE_FDATASYNC` defined here.

Some other solution seems desirable...

Reply:
Is this a NetBSD bug?

Reply:
This is not a NetBSD bug. The POSIX/SUSvX whatever it is now is being more liberally interpreted by Linux as supporting directory fdatasync, but arguably this is incorrect. NetBSD interprets conditions and/or functionality like this quite a bit more strictly than Linux does, and does so (somewhat) independently and without working to be compatible with Linux.